app/src/main/java/com/simon/harmonichackernews/network/NetworkComponent.java

```diff
@@ -0,0 +1,38 @@
+package com.simon.harmonichackernews.network;
+
+import android.content.Context;
+import android.os.Looper;
+
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+import com.simon.harmonichackernews.BuildConfig;
+
+import okhttp3.OkHttpClient;
+
+public class NetworkComponent {
+    private static volatile OkHttpClient okHttpClientInstance;
+    private static RequestQueue requestQueueInstance;
+
+    public static OkHttpClient getOkHttpClientInstance() {
+        if (okHttpClientInstance == null) {
+            synchronized (NetworkComponent.class) {
+                if (okHttpClientInstance == null) {
+                    okHttpClientInstance = new OkHttpClient();
```

Comment:
I've noticed that Harmonic doesn't send `User-Agent` header with any of its requests. Unless you have a good reason to do so, I'd recommend to add something like `Harmonic-HN/version` as user agent

Reply:
This is a reasonable take, did this in https://github.com/SimonHalvdansson/Harmonic-HN/pull/86/commits/fcd85f3a3ab81f1640198eab4e5d615b90098f07